### PR TITLE
fix(headless): add full path as UA category facet value

### PR DESCRIPTION
**Pull request by ThibodeauJF**
Tests on the way
https://coveord.atlassian.net/browse/KIT-579

### CODE REVIEW

File: packages/headless/src/features/facets/facet-set/facet-set-analytics-actions-utils.ts [hunk 1]
```diff
@@ -61,8 +61,14 @@ export function buildFacetSelectionChangeMetadata(
...
   const {facetId, facetValue} = payload;
   const base = buildFacetBaseMetadata(facetId, state);
-
-  return {...base, facetValue};
+  const facetType = getFacetType(state, facetId);
+  return {
+    ...base,
+    facetValue:
+      facetType === 'hierarchical'
+        ? getCategoryFacetSelectedValue(state, facetId)
```
Comment (samisayegh) [inline packages/headless/src/features/facets/facet-set/facet-set-analytics-actions-utils.ts:69]: We should add unit tests for this new code path,

File: packages/headless/src/features/facets/facet-set/facet-set-analytics-actions-utils.ts [hunk 2]
```diff
@@ -83,39 +89,54 @@ export const buildFacetStateMetadata = (
...
     );
+
+    if (facetType === 'hierarchical') {
+      const hasSelectedValue = !!categoryFacetSelectedValuesSelector(
+        state,
+        facetResponse.facetId
+      ).length;
+
+      if (!hasSelectedValue) {
+        return;
```
[the remark this answers is not in the document]
Comment (ThibodeauJF) [reply packages/headless/src/features/facets/facet-set/facet-set-analytics-actions-utils.ts:105]: It had the same condition before for all facet type when there were no selected value.
This method is for the facetState, which only contains selected values

File: packages/headless/src/features/facets/facet-set/facet-set-analytics-actions-utils.ts [hunk 3]
```diff
@@ -145,15 +166,39 @@ const mapRangeDisplayFacetValueToAnalytics = (
...
+  state: SectionNeededForFacetMetadata,
+  facetId: string
+) => {
+  const selectedCategoryFacetValues = categoryFacetSelectedValuesSelector(
+    state,
+    facetId
+  );
+  return selectedCategoryFacetValues
+    .map((selectedCategoryFacetValue) => selectedCategoryFacetValue.value)
+    .join(';');
```
[the remark this answers is not in the document]
Comment (ThibodeauJF) [reply packages/headless/src/features/facets/facet-set/facet-set-analytics-actions-utils.ts:190]: I don't think UA will differentiate & know which delimiter is used, it makes sense for it to be standardized, but I can validate

File: packages/headless/src/features/facets/facet-set/facet-set-analytics-actions-utils.ts [hunk 4]
```diff
@@ -145,15 +166,39 @@ const mapRangeDisplayFacetValueToAnalytics = (
...
+};
+
+const mapCategoryFacetValueToAnalytics = (
+  state: SectionNeededForFacetMetadata,
+  facetId: string
+) => {
+  const valuePosition = 1;
```
[the remark this answers is not in the document]
Comment (ThibodeauJF) [reply packages/headless/src/features/facets/facet-set/facet-set-analytics-actions-utils.ts:197]: If im not mistaken, the selected value of a category facet is always the first, when it's returned by the API we only see its children

**Review by olamothe — APPROVED** [review]

File: packages/headless/src/features/facets/facet-set/facet-set-analytics-actions-utils.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (samisayegh) [reply packages/headless/src/features/facets/facet-set/facet-set-analytics-actions-utils.ts:197]: I see, thanks

**Review by samisayegh — APPROVED** [review]

**Pull request merged**